tasks/log_groups.yml

```diff
@@ -20,8 +20,9 @@
       metric_name: "{{ filter.metric_transformation.metric_name }}"
       metric_namespace: "{{ aws_cwa_namespace }}"
       metric_value: "{{ filter.metric_transformation.metric_value }}"
-    aws_access_key: "{{ deploy_aws_access_key | default(aws_access_key | default(lookup('env','AWS_ACCESS_KEY_ID'))) }}"
-    aws_secret_key: "{{ deploy_aws_secret_key | default(aws_secret_key | default(lookup('env','AWS_SECRET_ACCESS_KEY'))) }}"
+    aws_access_key: "{{ aws_access_key | default(omit) }}"
+    aws_secret_key: "{{ aws_secret_key | default(omit) }}"
+    security_token: "{{ security_token }}" # Use token to allow for assumed roles
```

Comment:
I can't see any reason to hard code so many variables into a role for the same value. Also, env lookups shouldn't be done inside a role but instead, there should be fixed variables that are passed to the role.

I've also added `security_token` which I think might be needed for assumed roles but I'm not sure. Might be able to remove `security_token` if it turns out assumable roles are fine using access key and secret.

Reply:
I've changed these back to the module's expected variables.

Reply:
`security_token` is unlikely to be needed. any assumed role is likely to be set via env vars so should be picked up automatically.

As commented elsewhere the whole `{{ deploy_aws_access_key | default(aws_access_key | default(lookup('env','AWS_ACCESS_KEY_ID'))) }}"` is a mess and probably the defaults are better, but this is used in a bunch of places and needs removing across the board so better to do as a separate task

Reply:
I'm happy to remove `security_token` as I was never sure if it was needed. There was one example where it was needed in a different playbook but I think my inventory or envs were the issue there.

I'll also remove the omit as without `security_token` we want this to fail if the key and secrets are missing :)